udata/core/spatial/models.py

```diff
@@ -39,9 +59,43 @@ class GeoLevel(db.Document):
 
 class GeoZoneQuerySet(db.BaseQuerySet):
     def valid_at(self, valid_date):
-        compare_string = valid_date.isoformat()
-        return self(validity__end__gt=compare_string,
-                    validity__start__lt=compare_string)
+        '''Limit current QuerySet to zone valid at a given date'''
+        is_valid = db.Q(validity__end__gt=valid_date,
+                        validity__start__lte=valid_date)
+        no_validity = db.Q(validity=None)
+        return self(is_valid | no_validity)
+
+    def latest(self):
+        '''
+        Fetch the latest valid zone matching a QuerySet.
+
+        Ensuring the QuerySet unicity for (level, code)
+        is you responsibility.
+        '''
+        return self.order_by('-validity__end').first()
```

Comment:
I think we should check validity for today. It would prevent from returning a zone that is not anymore active/valid.

Reply:
That's the expected behavior: I want the latest zone wether it exists or not anymore.
First implementation was using `today` but testing revealed that it wasn't working for ended zones (which we have).